src/math/Interpolate.cc

```diff
@@ -157,7 +157,7 @@ ::gsl_interp_type const *styleToGslInterpType(Interpolate::Style const style) {
         case Interpolate::UNKNOWN:
             throw LSST_EXCEPT(pex::exceptions::InvalidParameterError,
                               "I am unable to make an interpolator of type UNKNOWN");
-        case Interpolate::NUM_STYLES:
+        default:
```

Comment:
Without a `default` clause GCC gives a warning about falling through to the end of the function even when every constant is mentioned (despite also doing the check you mentioned).

I think I have an idea that will satisfy both rules, let me check.